streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -481,10 +500,25 @@ object ZSink extends ZSinkPlatformSpecificConstructors {
     new ZSink(push) {}
 
   /**
-   * A sink that collects all of its inputs into a list.
+   * A sink that collects all of its inputs into a chunk.
    */
-  def collectAll[A]: ZSink[Any, Nothing, A, List[A]] =
-    foldLeftChunks(Chunk[A]())(_ ++ (_: Chunk[A])).map(_.toList)
+  def collectAll[A]: ZSink[Any, Nothing, A, Chunk[A]] =
+    //Cannot make a builder that initializes itself on the first operation.
+    // falling back to null instead of an option/None combination
+    foldLeftChunks[A, ChunkBuilder[A]](null: ChunkBuilder[A])((b, chunk) =>
```

Comment:
Can we just wrap the normal `ChunkBuilder.make()` in `ZIO.effectTotal`? That is going to be the easiest way to solve this?

Reply:
Then you are stuck with the `Sink[Any, Nohting,A, UIO[Chunkbuilder[A]]]` where you cannot get the Chunkbuilder out of the UIO

Reply:
@borissmidt
```scala
ZSink {
  for {
    builder <- UIO(ChunkBuilder.make()).toManaged_
    foldingSink = foldLeftChunks(...)
    push <- foldingSink.push
  } yield push
}
```